modules/admin_manual/pages/configuration/integration/ms-teams.adoc

```diff
@@ -0,0 +1,125 @@
+:publish-custom-app-url: https://docs.microsoft.com/en-us/MicrosoftTeams/upload-custom-apps
+:teams-app-setup-policies-url: https://docs.microsoft.com/en-us/microsoftteams/teams-app-setup-policies
+:manage-apps-url: https://docs.microsoft.com/en-us/MicrosoftTeams/manage-apps
+:what-are-tabs-url: https://docs.microsoft.com/en-us/microsoftteams/platform/tabs/what-are-tabs
+
+== Introduction
+
+If you're using Microsoft Teams in your organization or for private purposes, you will likely want to access your ownCloud installation from your Microsoft Teams account. For this purpose, we created the {appsource-url}[Microsoft AppSource] app *ownCloud Generator for Admins* with which you can generate a customized Microsoft Teams app for your users accessing your ownCloud services. Each ownCloud domain accessed requires a separate generated Microsoft Teams app for your users. The generated app will be available in your organization's app catalog.
+
+NOTE: As a prerequisite, the OpenID Connect app is required. If you already have a OpenID Connect configuration made with another service, you have to reconfigure with Microsoft Azure AD, as only one identity provider configuration is allowed.
+
+== Prerequisites
+
+To get this working, you need to install and/or configure four components:
+
+. Microsoft Azure Active Directory
+
+. ownCloud apps:
+.. {oc-marketplace-url}/apps/openidconnect[OpenID Connect] 
+.. https://github.com/owncloud/msteamsbridge[MS-Teams Bridge] 
+// (later available on marketplace or git only?)
+
+. The custom app(s) you have generated with *ownCloud Generator for Admins*
+
+. Microsoft Teams
+
+== ownCloud
+
+=== Installation
+
+Assuming you have an ownCloud server version 10.7 or higher already running in your company or for personal use, perform the following steps:
+
+. Install and enable the https://github.com/owncloud/msteamsbridge[MS-Teams Bridge] app, minimum required version: v1.0.0.
+// (later available on marketplace or git only?)
+
+. Install and enable the {oc-marketplace-url}/apps/openidconnect[OpenID Connect] app from the ownCloud marketplace, minimum required version: v2.0.0.
+
+=== Configure the MS-Teams Bridge App
+
+You need to configure the MS-Teams Bridge app in two steps:
+
+. Add a _header_ directive to the Apache `.htaccess` configuration located in your ownCloud web root in section `<IfModule mod_env.c>`
++  
+[source,apache]
+----
+Header merge Content-Security-Policy "frame-ancestors teams.microsoft.com *.teams.microsoft.com"
+----
++
+Using `merge`, the response header is appended to any existing header of the same name, unless the value to be appended already appears in the header's comma-delimited list of values. When a new value is merged onto an existing header it is separated from the existing header with a comma. Merge avoids that headers of the same type and content are sent multiple times. This can happen if headers are also set on other locations.
```

Comment:
```suggestion
Using `merge`, the response header is appended to any existing header of the same name, unless the value to be appended already appears in the header's comma-delimited list of values. When a new value is merged onto an existing header it is separated from the existing header with a comma. Merge avoids headers of the same type and content being sent multiple times. This can happen if headers are also set on other locations.
```